okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -639,16 +635,22 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
           val newStream = Http2Stream(streamId, this@Http2Connection, false, inFinished, headers)
           lastGoodStreamId = streamId
           streams[streamId] = newStream
-          listenerExecutor.execute("OkHttp $connectionName stream $streamId") {
-            try {
-              listener.onStream(newStream)
-            } catch (e: IOException) {
-              Platform.get().log("Http2Connection.Listener failure for $connectionName", INFO, e)
-              ignoreIoExceptions {
-                newStream.close(ErrorCode.PROTOCOL_ERROR, e)
+
+          // Use a different task queue for each stream because they should be handled in parallel.
+          val taskName = "OkHttp $connectionName stream $streamId"
+          taskRunner.newQueue(taskName).schedule(object : Task(taskName) {
+            override fun runOnce(): Long {
+              try {
+                listener.onStream(newStream)
+              } catch (e: IOException) {
+                Platform.get().log("Http2Connection.Listener failure for $connectionName", INFO, e)
+                ignoreIoExceptions {
+                  newStream.close(ErrorCode.PROTOCOL_ERROR, e)
+                }
               }
+              return -1L
```

Comment:
I wonder if there is a better way to hide this magic value?  e.g. 

```
Task.once {
   ...
}
```

Reply:
Probably. I’d like to get the behavior all good in the first pass, and then think about improving the syntax. This is internal-facing only so my preference is to be pretty frugal with fancy things.